README.md

```diff
@@ -32,17 +33,25 @@ You can configure the scope, which you pass in to the `provider` method via a `H
 
 * `scope`: A comma-separated list of permissions you want to request from the user. See [the Shopify API docs](http://docs.shopify.com/api/tutorials/oauth) for a full list of available permissions.
 
+* You **MUST** initiate the OmniAuth process by passing in a `shop` query parameter of the shop you’re requesting permissions for example. http://localhost:3000/auth/shopify?shop=example.myshopify.com. 
+
+* `setup`: You can provide a custom lambda which dynamically sets the site in the initializer 
```

Comment:
I agree it definitely might be confusing and shouldn't always be overriden but it should at least be apparent that you need to send the shop in the query string or the gem won't work at all. It might be a good idea to implement something in the error like `request.env['omniauth.errors']` that says `Please provide a valid shopify subdomain as the shop query parameter`.